src/sys/unix.rs

```diff
@@ -436,12 +475,17 @@ impl crate::Socket {
     /// Only supported on Apple platforms (`target_vendor = "apple"`).
     #[cfg(all(feature = "all", target_vendor = "apple"))]
     pub fn set_nosigpipe(&self, nosigpipe: bool) -> io::Result<()> {
+        self._set_nosigpipe(nosigpipe)
+    }
+
+    #[cfg(all(target_vendor = "apple"))]
+    pub(crate) fn _set_nosigpipe(&self, nosigpipe: bool) -> io::Result<()> {
         unsafe {
-            setsockopt::<c_int>(
+            setsockopt(
                 self.inner,
                 libc::SOL_SOCKET,
                 libc::SO_NOSIGPIPE,
-                nosigpipe as _,
+                nosigpipe as c_int,
```

Comment:
Let's leave this as `_`. Apple has a history of change system calls parameters, with the old code was forward compatible in that way.

Reply:
It's not a functional change, I just moved the `c_int` from the template parameter to the cast, in line with the rest of the code in this file.

We must be specific with the type here.